hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java

```diff
@@ -74,11 +73,6 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
         if (!op.hasTransactionID() || !op.transactionID().hasAccountID()) {
             throw new PreCheckException(INVALID_TRANSACTION_ID);
         }
-
-        // The receipt must exist for that transaction ID
-        final var recordCache = context.recordCache();
-        final var receipt = recordCache.getReceipt(op.transactionIDOrThrow());
-        mustExist(receipt, RECEIPT_NOT_FOUND);
```

Comment:
Why should we remove this ?

Reply:
In mono, we only return RECORD/RECEIPT_NOT_FOUND for the ANSWER_ONLY flow. If, in the modularized code, we validate this here before the findResponse method in either `NetworkTransactionGetRecordHandler` or `NetworkTransactionGetReceiptHandler`, it will cause this precheck to be thrown in COST_ANSWER, leading to a different behavior.

We already have a flow within the findResponse method that handles the cases for RECORD/RECEIPT_NOT_FOUND and returns it as the precheck for ANSWER_ONLY.